spacecmd/src/spacecmd/errata.py

```diff
@@ -92,31 +92,34 @@ def complete_errata_apply(self, text, line, beg, end):
     return self.tab_complete_errata(text)
 
 
-def do_errata_apply(self, args, only_systems=None):
+def do_errata_apply(self, args, errata=None, only_systems=None):
     arg_parser = get_argument_parser()
+    arg_parser.add_argument('errata', nargs="*")
     arg_parser.add_argument('-s', '--start-time')
 
     args, options = parse_command_arguments(args, arg_parser)
     only_systems = only_systems or []
+    errata = options.errata or errata
+    del options.errata
```

Comment:
I don't know much about `spacecmd`'s argument parsing, just two quick questions:

1. Can `$errata` exist in `args` *and* be passed as `errata`? If yes, why do we prefer what's in `args` over whats in `errata`?
2. I see `del` being used a bit in `spacewalk/`, but not everywhere. Can you elaborate on when we (should) use it to clear the `options` namespace?

Reply:
Thanks for the review Alex :+1:

Let me try to answer your questions:

  1. There shouldn't be a case when both are passed. The `errata` and `only_installed` parameters for this function are only passed when the function is called by `system_erratapply`, and in this case, we want the value coming from `errata`. When this function is directly called by the user via CLI, then errata is contained as part of `args` only. So, I can actually even use `errata = errata or options.errata` to be aligned with what we do for `only_installed`.
  
  2. Well, the thing is that the current way for `spacecmd` to check whether it is running in interactive mode or not is really bad, it basically assumes running non-interactive if there is any non-empty values passed inside `options`. This makes a bit tricky to deal with arguments passed from other spacemd functions. So, in this case, we don't want to directly assume that we run in non-interactive mode when we are passing `errata` in `args`.

I'll align what I just mentioned on 1.